### PR TITLE
feat: [IOBP-437] Add new wallet payment outcome error handling

### PR DESCRIPTION
**Pull request by mastro993**
⚠️ Depends on https://github.com/pagopa/io-app/pull/5378

## Short description
This PR adds the handling of the payment error outcomes.

## List of changes proposed in this pull request
- Added components mapping based on the outcome in `WalletPaymentOutcomeScreen`
- Added required locales keys
- Refactored `usePaymentFailureSupportModal` to remove the requirement of passing `RptId` as its argument.

## How to test
Within the **Profile > New Wallet > Payment** playground, using the `io-dev-api-server`, try to simulate different payment outcomes and check that the screen and its components are displayed correctly.

## Preview

https://github.com/pagopa/io-app/assets/6160324/cb7c33f1-9c0d-471c-accf-a362efc42264

### CODE REVIEW

File: ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const WalletPaymentFeebackBanner = () => {
+  const bannerViewRef = React.useRef<View>(null);
+
+  const handleBannerPress = () => {
+    void mixpanelTrack("VOC_USER_EXIT", {
+      screen_name: "PAYMENT_OUTCOMECODE_MESSAGE"
+    });
+
+    return openAuthenticationSession(
+      "https://io.italia.it/diccilatua/ces-pagamento",
```
Comment (Hantex9) [inline ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx:17]: What do you think about adding this URL into a constant file?

File: ts/features/walletV3/payment/hooks/usePaymentFailureSupportModal.tsx [hunk 2]
```diff
@@ -117,30 +122,21 @@ const usePaymentFailureSupportModal = ({
...
-            onPress={() =>
-              Linking.openURL(`tel:${PAGOPA_SUPPORT_PHONE_NUMBER}`)
-            }
-            variant="primary"
-            icon="phone"
-          />
-        )}
+        <ListItemAction
+          label={I18n.t("wallet.payment.support.phone", {
+            phoneNumber: "06.4520.2323"
```
Comment (Hantex9) [inline ts/features/walletV3/payment/hooks/usePaymentFailureSupportModal.tsx:131]: I think it is better to leave it as it was declaring the `PAGOPA_SUPPORT_PHONE_NUMBER`, maybe a merge conflict? What do you think?

**Review by Hantex9 — CHANGES_REQUESTED** [review]

File: ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx:17]: I had already moved this link to a const, IDK what happened, maybe a merge gone wrong. I'll fix it asap!

File: ts/features/walletV3/payment/hooks/usePaymentFailureSupportModal.tsx (hunk 2 above)
Comment (mastro993) [reply ts/features/walletV3/payment/hooks/usePaymentFailureSupportModal.tsx]: Fixed in https://github.com/pagopa/io-app/pull/5390/commits/336a54bf871c389fcea39cc1b0be327fb48a3137

File: ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx (hunk 1 above)
Comment (mastro993) [reply ts/features/walletV3/payment/components/WalletPaymentFeedbackBanner.tsx]: Fixed in https://github.com/pagopa/io-app/pull/5390/commits/336a54bf871c389fcea39cc1b0be327fb48a3137

**Review by Hantex9 — APPROVED** [review]
LGTM

**Pull request merged**